src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -1885,6 +1926,140 @@ class module_sram : public module_impl
     return payload;
   }
 
+  struct dlib_handle
+  {
+    void* handle = nullptr;
+
+    dlib_handle(const std::string& path)
+    {
+      handle = xrt_core::dlopen(path.c_str(), RTLD_LAZY);
+    }
+
+    ~dlib_handle()
+    {
+      if (handle)
+        xrt_core::dlclose(handle);
+    }
+  };
+
+  struct dtrace_util
+  {
+    std::unique_ptr<dlib_handle> lib_hdl;
+    std::string ctrl_file_path;
+    std::string map_data;
+  };
+
+  static bool
+  init_dtrace_helper(const module_impl* parent, dtrace_util& dtrace_obj)
```

Comment:
Return codes are not usually desired, but in this case make sense after reading how it is used.  This deserves a comment.  Basically, the exception alternative is not good when the success path allows init_dtrace_helper to be a no-op.